code/messaging/src/main/java/com/adobe/marketing/mobile/messaging/ContentCardEventObserver.kt

```diff
@@ -0,0 +1,184 @@
+/*
+  Copyright 2024 Adobe. All rights reserved.
+  This file is licensed to you under the Apache License, Version 2.0 (the "License");
+  you may not use this file except in compliance with the License. You may obtain a copy
+  of the License at http://www.apache.org/licenses/LICENSE-2.0
+  Unless required by applicable law or agreed to in writing, software distributed under
+  the License is distributed on an "AS IS" BASIS, WITHOUT WARRANTIES OR REPRESENTATIONS
+  OF ANY KIND, either express or implied. See the License for the specific language
+  governing permissions and limitations under the License.
+*/
+
+package com.adobe.marketing.mobile.messaging
+
+import com.adobe.marketing.mobile.MessagingEdgeEventType
+import com.adobe.marketing.mobile.aepcomposeui.interactions.UIEvent
+import com.adobe.marketing.mobile.aepcomposeui.observers.AepUIEventObserver
+import com.adobe.marketing.mobile.aepcomposeui.uimodels.SmallImageTemplate
+import com.adobe.marketing.mobile.aepcomposeui.utils.UIAction
+import com.adobe.marketing.mobile.services.Log
+
+class ContentCardEventObserver(private var callback: ContentCardCallback?) : AepUIEventObserver {
+    companion object {
+        private const val SELF_TAG = "ContentCardEventObserver"
+    }
+
+    override fun onEvent(event: UIEvent<*, *>) {
+        if (callback == null) {
+            callback = DefaultContentCardCallback()
```

Comment:
Why do we need the `DefaultContentCardCallback`?

Reply:
I've removed this as there are no longer boolean values being returned